boot/assets.go

```diff
@@ -182,12 +182,49 @@ func TrustedAssetsInstallObserverForModel(model *asserts.Model, gadgetDir string
 	if gadgetDir == "" {
 		return nil, fmt.Errorf("internal error: gadget dir not provided")
 	}
+	// TODO:UC20: clarify use of empty rootdir when getting the lists of
+	// managed and trusted assets
+	runBl, runTrusted, runManaged, err := gadgetMaybeTrustedBootloaderAndAssets(gadgetDir, "",
+		&bootloader.Options{
+			Role:        bootloader.RoleRunMode,
+			NoSlashBoot: true,
+		})
+	if err != nil {
+		return nil, err
+	}
+	// and the recovery bootloader, seed is mounted during install
+	seedBl, seedTrusted, _, err := gadgetMaybeTrustedBootloaderAndAssets(gadgetDir, InitramfsUbuntuSeedDir,
+		&bootloader.Options{
+			Role: bootloader.RoleRecovery,
+		})
+	if err != nil {
+		return nil, err
+	}
+	if !useEncryption {
+		// we do not care about trusted assets when not encrypting data
+		// partition
+		runTrusted = nil
+		seedTrusted = nil
+	}
+	hasManaged := len(runManaged) > 0
+	hasTrusted := len(runTrusted) > 0 || len(seedTrusted) > 0
+	if !hasManaged && !hasTrusted {
```

Comment:
maybe it relates to your wondering today, but given this is also used to track the actual key, I think we should also check && !useEncryption here?

Reply:
I think it's fine to add it for now. With #9476 there's no more SetEncryptionKey.

Reply:
Pushed